xmtp_api_grpc_gateway/tests/wasm_test.rs

```diff
@@ -129,3 +135,53 @@ pub async fn test_batch_query_publish_batch_query() {
     assert_eq!(vec![1, 1, 1, 1], e1.message);
     assert_eq!(vec![2, 2, 2, 2], e2.message);
 }
+
+#[wasm_bindgen_test]
+pub async fn test_client_subscribe() {
+    log("test_client_subscribe");
+    let xmtp_url: String = "http://localhost:5555".to_string();
+    let api = XmtpGrpcGatewayClient::new(xmtp_url);
+
+    let topic = uuid::Uuid::new_v4();
+    log(&topic.to_string());
+    let auth_token = ""; // TODO
+
+    // Subscribe to a topic and then expect to see what we publish.
+    let stream = api
+        .subscribe(SubscribeRequest {
+            content_topics: vec![topic.to_string()],
+            // ..SubscribeRequest::default()
+        })
+        .await
+        .expect("subscribed");
+
+    // This is how the subscription task tells the foreground task that it got a message.
+    let (tx, rx) = oneshot::channel();
+
+    wasm_bindgen_futures::spawn_local(async move {
+        pin_mut!(stream);
+        let env = stream.next().await.expect("received message");
+        tx.send(env).expect("notified that we got the message");
+    });
+
+    log("publishing message");
+    api.publish(
+        auth_token.to_string(),
+        PublishRequest {
+            envelopes: vec![Envelope {
+                content_topic: topic.to_string(),
+                message: vec![1, 2, 3, 4],
+                timestamp_ns: 1234,
+            }],
+        },
+    )
+    .await
+    .expect("published");
+
+    if let Ok(env) = rx.await {
```

Comment:
Nooby question, but what is the point of the `channel` here? If we are `awaiting` anyway, what difference does it make if we do `rx.await` rather than `stream.next().await` directly?